### ISSUE TITLE
8$: Error on await () => {}

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#279]
Within async functions, `await () => {}` is invalid syntax, at least according to Node.js v15.5.1:
```javascript
async function g() {
  f(await () => {});
}
g();
```
```
hello.js:2
  f(await () => {});
    ^^^^^^^

SyntaxError: Malformed arrow function parameter list
```

quick-lint-js reports no error for this code. Assuming that Node.js is standards-conforming, quick-lint-js should report an error for an attempted await for an arrow function.

**Comment by strager** [comment]
#278 is a related issue. #278 is about `async()=>{}` outside async functions. This task is about `async()=>{}` inside async functions. Perhaps the end result should be the same (i.e. same diagnostic and error recovery), but the parsing code seems to be different, so I thought they deserved separate tasks.

**Comment by keyehzy** [comment]
Will try this one, PR'ing soon.
